implementations/rust/ockam/ockam_command/src/vault/util.rs

```diff
@@ -0,0 +1,32 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Ok;
+use ockam_api::config::cli;
+use ockam_vault::storage::FileStorage;
+use ockam_vault::Vault;
+use slug::slugify;
+
+pub fn check_if_vault_exists(path: PathBuf) -> bool {
+    path.as_path().exists()
+}
+
+pub async fn create_new_vault(vault_name: String) -> Result<(), anyhow::Error> {
+    let path = get_config_dir().join(slugify(&format!("vault-{}", vault_name)));
+    if check_if_vault_exists(path.clone()) {
+        return Err(anyhow::anyhow!("Vault Exists Already!"));
+    }
+    tokio::fs::create_dir_all(path.as_path()).await?;
+    let path = path.join("vault.json");
+
+    let storage = FileStorage::create(path.clone()).await?;
+    let _ = Vault::new(Some(Arc::new(storage)));
+
+    Ok(())
+}
```

Comment:
This function is only used by the `create` command, so it should be in the `create.rs` file.